modules/terraform/output.go

```diff
@@ -49,6 +50,143 @@ func OutputRequiredE(t testing.TestingT, options *Options, key string) (string,
+
+	for k, v := range m {
+		switch vt := v.(type) {
+		case map[string]interface{}:
+			nestedMap, err := parseMap(vt)
+			if err != nil {
+				return nil, err
+			}
+			result[k] = nestedMap
+		case []interface{}:
+			nestedList, err := parseListOfMaps(vt)
+			if err != nil {
+				return nil, err
+			}
+			result[k] = nestedList
+		case float64:
+			testInt, err := strconv.ParseInt((fmt.Sprintf("%v", vt)), 10, 0)
+			if err == nil {
+				result[k] = int(testInt)
+			} else {
+				result[k] = vt
+			}
+		default:
+			result[k] = vt
+		}
+
+	}
+
+	return result, nil
+}
+
+// OutputMapOfObjects calls terraform output for the given variable and returns its value as a map of lists/maps.
+// If the output value is not a map of lists/maps, then it fails the test.
+func OutputMapOfObjects(t testing.TestingT, options *Options, key string) map[string]interface{} {
+	out, err := OutputMapOfObjectsE(t, options, key)
+	require.NoError(t, err)
+	return out
+}
+
+// OutputMapOfObjectsE calls terraform output for the given variable and returns its value as a map of lists/maps.
+// Also returns an error object if an error was generated.
+// If the output value is not a map of lists/maps, then it fails the test.
+func OutputMapOfObjectsE(t testing.TestingT, options *Options, key string) (map[string]interface{}, error) {
+	out, err := RunTerraformCommandAndGetStdoutE(t, options, "output", "-no-color", "-json", key)
+
+	if err != nil {
+		return nil, err
+	}
+
+	var output map[string]interface{}
+
+	if err := json.Unmarshal([]byte(out), &output); err != nil {
+		return nil, err
+	}
+
+	var result map[string]interface{}
+
+	result, err = parseMap(output)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return result, nil
+}
+
+// OutputListOfObjects calls terraform output for the given variable and returns its value as a list of maps/lists.
+// If the output value is not a list of maps/lists, then it fails the test.
+func OutputListOfObjects(t testing.TestingT, options *Options, key string) []map[string]interface{} {
+	out, err := OutputListOfObjectsE(t, options, key)
+	require.NoError(t, err)
+	return out
+}
+
+// OutputListOfObjectsE calls terraform output for the given variable and returns its value as a list of maps/lists.
+// Also returns an error object if an error was generated.
+// If the output value is not a list of maps/lists, then it fails the test.
+func OutputListOfObjectsE(t testing.TestingT, options *Options, key string) ([]map[string]interface{}, error) {
+	out, err := RunTerraformCommandAndGetStdoutE(t, options, "output", "-no-color", "-json", key)
+
+	if err != nil {
+		return nil, err
+	}
+
+	var output []map[string]interface{}
+
+	if err := json.Unmarshal([]byte(out), &output); err != nil {
+		return nil, err
+	}
+
+	var result []map[string]interface{}
+
+	for _, m := range output {
+		newMap, err := parseMap(m)
+
+		if err != nil {
+			return nil, err
+		}
+
+		result = append(result, newMap)
+	}
+
+	return result, nil
```

Comment:
```suggestion
	return parseListOfMaps(output)
```

Reply:
This part is actually needed to change the type from `[]interface` to `[]map[string]interface`.

Reply:
Isn't that exactly what `parseListOfMaps` does?

Reply:
no

```$ go test . -timeout 30m -run TestOutputListOfObjects
# github.com/gruntwork-io/terratest/modules/terraform [github.com/gruntwork-io/terratest/modules/terraform.test]
./output.go:179:24: cannot use output (type []map[string]interface {}) as type []interface {} in argument to parseListOfMaps
FAIL    github.com/gruntwork-io/terratest/modules/terraform [build failed]
FAIL```

Reply:
Gah. Go and it's lack of generics.